be/src/storage/binlog_manager.cpp

```diff
@@ -43,9 +43,195 @@ BinlogManager::~BinlogManager() {
     }
 }
 
+Status BinlogManager::init(BinlogLsn min_valid_lsn, std::list<int64_t>& sorted_valid_versions) {
+    // 1. list all of binlog files
+    std::list<int64_t> binlog_file_ids;
+    Status status = BinlogUtil::list_binlog_file_ids(_path, &binlog_file_ids);
+    if (!status.ok()) {
+        std::string errMsg = fmt::format("Failed to init binlog because of list error, tablet {}, status: {}",
+                                         _tablet_id, status.to_string());
+        LOG(ERROR) << errMsg;
+        _init_failure.store(true);
+        return Status::InternalError(errMsg);
+    }
+    binlog_file_ids.sort();
+
+    // 2. recover binlog from the largest version to the smallest version
```

Comment:
why reverse

Reply:
file metas are pushed back to `recovered_file_metas` in the descending order of file ids, and we want to recover them in the ascending order, so reverse the vector first